apps/AppWithWearable/lib/backend/auth.dart

```diff
@@ -127,3 +127,18 @@ Future<void> signOut(BuildContext context) async {
   }
   // context.pushReplacementNamed('auth');
 }
+
+listenAuthStateChanges() {
+  FirebaseAuth.instance.authStateChanges().listen((User? user) {
+    if (user == null) {
+      debugPrint('User is currently signed out!');
+      SharedPreferencesUtil().onboardingCompleted = false;
```

Comment:
Should sign out mean the user needs to do onboarding again?

Reply:
Well not really, but given the onboarding flow only involves notification permission and connecting device, I guess we can start the onboarding flow again.